File: BOJ/C++/1463-1로만들기.cpp
#include <iostream>

int calc(int num, int depth, int *minResult) {
    if (depth >= *minResult) {
        return 0;
    }
    if(num == 1) {
        *minResult = depth;
    }
    if(num%3 == 0) {
        calc(num/3, depth+1, minResult);
    }
    if(num%2 == 0) {
        calc(num/2, depth+1, minResult);
    }
    return calc(num-1, depth + 1, minResult);
}

int main() {
    int x, minResult = 100000000;
    std::cin >> x;
    calc(x, 0, &minResult);
    std::cout << minResult;
    return 0;
}

File: BOJ/Python/1463-1로만들기.py
x = int(input())
result = []

def calc(num, depth = 0):
    global result
    if num == 1:
        result.append(depth)
        return
    if num%3 == 0:
        calc(num//3, depth + 1)
    if num%2 == 0:
        calc(num//2, depth + 1)
    return calc(num -1, depth + 1)

calc(x)
print(min(result))
